### ISSUE TITLE
istio can not work well with erlang application

### ISSUE THREAD

**Issue by eastpav** [istio/istio#6029]
My kubernetes was deployed by kubespray 2.5.0 with istio auto-inject on. Everything is fine but EMQ. I tried that run EMQ manually in pod with elrang node name 'emq@127.0.0.1' , it works ok.(work ok means that I can use command 'emqttd_ctl status' get the application's status). unfortunately, it works not ok with the node name 'emq@10.233.98.142'(10.233.98.142 is pod's ip address), command 'emqtt_ctl status' response that 
```
Node 'emq@10.233.98.142' not responding to pings.
```
PS: if no istio, EMQ works fine.I think, not only EMQ, this is a problem with erlang application.
if node not responding to ping, EMQ can not build a cluster. How can I fix this? or additional configuration?

**Comment by ymesika** [comment]
Out of curiosity, why don't you have a Service and use its name instead of the Pod's IP?

**Comment by eastpav** [comment]
It is the same problem to use Service  instead of Pod's IP.
I found that erlang node cannot register infomation to EPMD, hence, node can not responding to pings. 
In my other k8s without istio, the response of execute 'erl_epmd:names().' in erlang shell like this:
```
{ok,[{"emq",6369},{"ttt",44614}]}
```
and, in k8s with istio, like this:
```
{ok,[]}
```

**Comment by ymesika** [comment]
I'm not sure if it's related to your specific issue but did you add the EPMD port to the service?
See #5989

**Comment by eastpav** [comment]
You mean the EPMD port also must be exposed? istio will manage all of the socket traffic inside and outside Pod, right?

**Comment by esnible** [comment]
Run `netstat -tulpn` and make sure the *Local Address* is 0.0.0.0 or 127.0.0.0 for all listeners.  I don't know about Erlang, but applications that explicitly listen on their external address (e.g. 10.233.98.142) will not work under Istio.  This is because the Envoy proxy traffic does not come from off-pod.

**Comment by howardjohn** [comment]
Is this still an issue? I have rabbitmq (erlang app) running on Istio 1.1.

**Comment by k0da** [comment]
@howardjohn Yes, I have istio 1.1.3 deployed and I have to define a service for an erlang app describing 4369 and 50000 ports in order to have ping within pod to work
